Dependencies/AndroidExtensions/Include/AndroidExtensions/JavaWrappers.h

```diff
@@ -76,16 +76,33 @@ namespace java::lang
         jbyteArray m_byteArray;
     };
 
+    class Class
+    {
+    public:
+        Class(const char* className);
+        Class(const jclass classObj);
+
+        operator jclass() const;
+
+        bool IsAssignableFrom(Class otherClass);
+
+    protected:
+        JNIEnv* m_env;
+        const jclass m_class;
+    };
+
     class Object
     {
     public:
         operator jobject() const;
+        Class Class();
 
     protected:
-        Object(const char* className, jobject object);
+        Object(const char* className);
+        Object(jobject object);
 
         JNIEnv* m_env;
-        const jclass m_class;
+        const class Class m_class;
```

Comment:
I don't think you need/want the `const class` part of this. I think it can just be `Class m_class`. @bghgary?

Reply:
const seems okay to me. The `class` part is extraneous though.

```suggestion
        const Class m_class;
```

Reply:
The class identifier was required in this case because I was also adding a method to the Object wrapper named `Class()`. I've renamed the method to GetClass to more accurately match the Java syntax which also means I can get rid of the class identifier here.